synapse/storage/databases/main/room.py

```diff
@@ -213,20 +213,28 @@ async def store_room(
             logger.error("store_room with room_id=%s failed: %s", room_id, e)
             raise StoreError(500, "Problem creating room.")
 
-    async def get_room(self, room_id: str) -> Optional[Dict[str, Any]]:
+    async def get_room(self, room_id: str) -> Optional[Tuple[bool, str, bool]]:
         """Retrieve a room.
 
         Args:
             room_id: The ID of the room to retrieve.
         Returns:
-            A dict containing the room information, or None if the room is unknown.
+            A tuple containing the room information:
+                * True if the room is public
+                * The room creator
+                * True if the room has an auth chain index
+
+            or None if the room is unknown.
         """
-        return await self.db_pool.simple_select_one(
-            table="rooms",
-            keyvalues={"room_id": room_id},
-            retcols=("room_id", "is_public", "creator", "has_auth_chain_index"),
-            desc="get_room",
-            allow_none=True,
+        return cast(
+            Optional[Tuple[bool, str, bool]],
+            await self.db_pool.simple_select_one(
```

Comment:
Is this going to give us integers on sqlite rather than bools?

Reply:
Also: the boolean fields seem to be nullable in the DB. Sorry if there is already an issue for this.

Reply:
Filed #16614.

Reply:
> Is this going to give us integers on sqlite rather than bools?

Luckily in this case the we can just do `bool(...)` to handle both ints and nulls. Turns out we don't ever use the creator, so I dropped it from the response.